core/validatorapi/validatorapi.go

```diff
@@ -309,11 +313,18 @@ func (c Component) SubmitBeaconBlock(ctx context.Context, block *spec.VersionedS
 		return err
 	}
 
-	pubkey, err := c.awaitProposerFunc(ctx, int64(slot))
+	// Get proposer pubkey (this is a blocking query).
+	dutySet, err := c.getDutyFunc(ctx, core.Duty{Slot: int64(slot), Type: core.DutyProposer})
 	if err != nil {
 		return err
 	}
 
+	// There should be single duty proposer for the slot
+	var pubkey core.PubKey
+	for pk := range dutySet {
+		pubkey = pk
+	}
```

Comment:
could maybe extract this a reuable method (it is used in two places and quite verbose). Also, it isn't blocking anymore

```
pubkey, err := c.getProposerPubkey(ctx, slot)
```

Reply:
done